src/api/app/controllers/webui/package_controller.rb

```diff
@@ -109,6 +109,23 @@ def show
 
     @services = @files.any? { |file| file[:name] == '_service' }
 
+    download_name_candidates = []
+    binary_list = Xmlhash.parse(Backend::Api::Build::Project.binarylist(@project))
```

Comment:
@seto49 I think its possible to ask the backend directly for the binarylist of a specific package, but seems to be not implemented in our API wrapper. I will check what we could do here, but it would safe some iteration here in the code

Reply:
Yes, I found a way to limit the result to a specific package, asking for all packages of a certain project will lead to performance problems, imagine a project like openSUSE:Factory, I will push a commit to the PR that allows you to ask the backend for a specific package

Reply:
Wasnt able to add it to your PR, so I created a seperate one. Incase it gets merged you can use it here https://github.com/openSUSE/open-build-service/pull/10757

Reply:
@krauselukas thank you very much! I totally agree that my implementation was wasteful. I am new to the project and didn't want to "mess with" the backend just yet. I'll rebase this PR once your code is merged.